### PR TITLE
Feature/ATG-106 Unit testing

### PR DESCRIPTION
**Pull request by IraCherginyshka**

### CODE REVIEW

File: src/app/components/game-control-option/game-control-option.component.spec.ts [hunk 1]
```diff
@@ -93,4 +94,101 @@ describe('GameControlOptionComponent', () => {
 
     expect(component.activeField).toBe('play');
   });
+
+  it('should save controls in localStorage if they are equal to the default controls by calling saveControls function', () => {
+    spyOn(component, 'saveControls').and.callThrough();
```
Comment (oleksandrstarov) [inline src/app/components/game-control-option/game-control-option.component.spec.ts]: What is the reason to use spy here?

File: src/app/components/game-control-option/game-control-option.component.spec.ts [hunk 2]
```diff
@@ -93,4 +94,101 @@ describe('GameControlOptionComponent', () => {
...
+      right: 'ArrowRight',
+    };
+
+    component.controls = customControls;
+    component.saveControls();
+    expect(localStorage.getItem(LocalStorage.CONTROLS)).toEqual(JSON.stringify(customControls));
+  });
+
+  it('should not save controls in localStorage if they are equal to the default controls by calling saveControls function', () => {
+    spyOn(component, 'saveControls').and.callThrough();
```
Comment (oleksandrstarov) [inline src/app/components/game-control-option/game-control-option.component.spec.ts]: Same as above

File: src/app/components/game-control-option/game-control-option.component.spec.ts [hunk 3]
```diff
@@ -93,4 +94,101 @@ describe('GameControlOptionComponent', () => {
...
+      [ControlsEnum.RIGHT]: DefaultSettings.RIGHT,
+    };
+
+    component.controls = defaultControls;
+    component.saveControls();
+    expect(localStorage.getItem(LocalStorage.CONTROLS)).toBeNull();
+  });
+
+  it('should not set a new keyboard to controls if not change isSetMode', () => {
+    spyOn(component, 'keyBoardInput').and.callThrough();
```
Comment (oleksandrstarov) [inline src/app/components/game-control-option/game-control-option.component.spec.ts]: same as above

File: src/app/components/game-control-option/game-control-option.component.spec.ts [hunk 4]
```diff
@@ -93,4 +94,101 @@ describe('GameControlOptionComponent', () => {
...
+
+    const event = new KeyboardEvent('keypress', {
+      key: 'f',
+    });
+    fixture.nativeElement.dispatchEvent(event);
+
+    component.keyBoardInput(event);
+
+    expect(component.controls.right).not.toBe('f');
```
Comment (oleksandrstarov) [inline src/app/components/game-control-option/game-control-option.component.spec.ts:134]: Good job!

File: src/app/components/game-control-option/game-control-option.component.spec.ts [hunk 5]
```diff
@@ -93,4 +94,101 @@ describe('GameControlOptionComponent', () => {
...
+  });
+
+  it('should set a new keyboard to controls if it is valid and unique', () => {
+    spyOn(component, 'setControl').and.callThrough();
+    spyOn(component, 'keyBoardInput').and.callThrough();
```
Comment (oleksandrstarov) [inline src/app/components/game-control-option/game-control-option.component.spec.ts]: No need to use spy, you can simply call these functions

**Review by oleksandrstarov — CHANGES_REQUESTED** [review]

File: src/app/components/game-control-option/game-control-option.component.spec.ts (hunk 1 above)
Comment (IraCherginyshka) [reply src/app/components/game-control-option/game-control-option.component.spec.ts]: Fixed

File: src/app/components/game-control-option/game-control-option.component.spec.ts (hunk 2 above)
Comment (IraCherginyshka) [reply src/app/components/game-control-option/game-control-option.component.spec.ts]: Fixed

File: src/app/components/game-control-option/game-control-option.component.spec.ts (hunk 3 above)
Comment (IraCherginyshka) [reply src/app/components/game-control-option/game-control-option.component.spec.ts]: Fixed

File: src/app/components/game-control-option/game-control-option.component.spec.ts (hunk 5 above)
Comment (IraCherginyshka) [reply src/app/components/game-control-option/game-control-option.component.spec.ts]: I understanded.
Thanks!

**Review by oleksandrstarov — APPROVED** [review]

**Pull request merged**